src/logworker.cpp

```diff
@@ -120,6 +122,14 @@ namespace g3 {
       return std::unique_ptr<LogWorker>(new LogWorker);
    }
 
+   void LogWorker::removeWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
+      auto bg_removesink_call = [this, &sink] {
```

Comment:
pass by reference should never be needed. Adding and removing sinks is not high performant code so I am still waiting for proof that this is an issue. 

If it only surfaces on some type of systems then please provide that information and I will look into it.